lib/Logger.ts

```diff
@@ -55,49 +45,46 @@ type Loggers = {
 
 class Logger {
   private level: string;
-  private logDir: string;
   private context: Context;
   private logger: any;
   private instanceId: number;
 
-  private static defaultLogDir = 'logs';
-
-  constructor({ level, instanceId, logDir, context }: {level: string, instanceId?: number, logDir?: string, context: Context}) {
+  constructor({ level, file, context, instanceId }: {level: string, file: string, context: Context, instanceId?: number}) {
     this.level = level;
-
-    this.logDir = logDir || Logger.defaultLogDir;
     this.context = context || Context.GLOBAL;
     this.instanceId = instanceId || 0;
 
-    if (!fs.existsSync(this.logDir)) {
-      fs.mkdirSync(this.logDir);
+    const transports = [
+      new winston.transports.Console({
+        level: this.level,
+        colorize: true,
+        format: this.getLogFormat(true),
+      }),
+    ];
+
+    if (file !== '') {
+      transports.push(new winston.transports.File({
+        level: this.level,
+        filename: file,
+        format: this.getLogFormat(false),
+      }));
     }
+
     this.logger = winston.createLogger({
+      transports,
       levels: LevelPriorities,
-      transports: [
-        new winston.transports.Console({
-          level: this.level,
-          colorize: true,
-          format: this.getLogFormat(true),
-        }),
-        new winston.transports.File({
-          level: this.level,
-          filename: path.join(this.logDir, contextFileMap[this.context]),
-          format: this.getLogFormat(false),
-        }),
-      ],
     });
   }
 
-  public static createLoggers = (level: string, instanceId = 0): Loggers => {
+  public static createLoggers = (level: string, file = '', instanceId = 0): Loggers => {
```

Comment:
Here too let's rename `file` to `filename` so we use the same name as the winston config option.